include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -678,24 +693,41 @@ class FEEvaluationBase
   void read_dof_values_plain (const VectorType *src_data[]);
 
   /**
-   * Internal data fields that store the values. Derived classes will know the
-   * length of all arrays at compile time and allocate the memory on the
-   * stack. This makes it possible to cheaply set up a FEEvaluation object and
-   * write thread-safe programs by letting each thread own a private object of
-   * this type. In this base class, only pointers to the actual data are
-   * stored.
-   *
+   * This is the general array for all data fields.
+   */
+  AlignedVector<VectorizedArray<Number> > *scratch_data_array;
```

Comment:
Yes, it indeed matches with the other types. The other reason for using a plain pointer here is that this data field usually comes from a thread local storage pool, so to hand it back and mark it as available again, I do not see a better solution than the plain pointer. This is also the way we use the list of objects in `WorkStream`. I will add a note that this usage is similar to how `WorkStream` handles the scratch data.